packages/store/addon/-private/system/core-store.ts

```diff
@@ -1436,9 +1439,20 @@ abstract class CoreStore extends Service {
     }
     const type = normalizeModelName(modelName);
     const normalizedId = ensureStringId(id);
-    const resource = constructResource(type, normalizedId);
+    const resource: ResourceIdentifierObject = constructResource(type, normalizedId);
```

Comment:
this type/id should be a recordIdentifer not a resource

Reply:
Sounds like we need to adjust the return type of `constructResource` to `ResourceIdentifierObject | RecordIdentifier`?